src/Symfony/Component/Form/ChoiceList/Factory/DefaultChoiceListFactory.php

```diff
@@ -48,13 +48,16 @@ public function createListFromLoader(ChoiceLoaderInterface $loader, $value = nul
     public function createView(ChoiceListInterface $list, $preferredChoices = null, $label = null, $index = null, $groupBy = null, $attr = null)
     {
         $preferredViews = [];
+        $preferredViewsOrder = [];
         $otherViews = [];
         $choices = $list->getChoices();
         $keys = $list->getOriginalKeys();
 
         if (!\is_callable($preferredChoices) && !empty($preferredChoices)) {
-            $preferredChoices = function ($choice) use ($preferredChoices) {
-                return \in_array($choice, $preferredChoices, true);
+            // make sure we have keys that reflect order
+            $preferredChoices = array_values($preferredChoices);
+            $preferredChoices = static function ($choice) use ($preferredChoices) {
```

Comment:
Hello, I'm new to reviews in symfony. Should the usage of `static` be considered a BC break? It's unlikely someone relies on a Factory in preferredChoices, but possible.

Reply:
This closure is only used if the user didn't configure the `preferred_choices` option as a callable (but as an array). And since we do not access the surrounding object context this change is fine.